python/kserve/kserve/protocol/dataplane.py

```diff
@@ -224,11 +227,22 @@ def model_ready(self, model_name: str) -> bool:
 
         return self._model_registry.is_model_ready(model_name)
 
-    def decode(self, body, headers) -> Tuple[Union[Dict, InferRequest], Dict]:
+    def decode(
+        self,
+        body,
+        headers,
+        protocol_version: str = PredictorProtocol.REST_V1.value,
+        model_name: str = None,
+    ) -> Tuple[Union[Dict, InferRequest], Dict]:
         t1 = time.time()
         attributes = {}
         if isinstance(body, InferRequest):
             return body, attributes
+        elif isinstance(body, InferenceRequest) or (
```

Comment:
I remember `InferenceRequest` was always converted to `InferRequest` before hitting here